### ISSUE TITLE
Possible to get this working with a Virtual List?

### ISSUE THREAD

**Issue by saturnonearth** [isaacHagoel/svelte-dnd-action#432]
Is it possible to get this working with a virtual list such as [https://www.npmjs.com/package/svelte-tiny-virtual-list](https://www.npmjs.com/package/svelte-tiny-virtual-list)

**Comment by isaacHagoel** [comment]
I am pretty sure it should be possible, the question is how hard.
How do you envision this working (as far as the UX goes)? more items
appearing as you drag lower and lower down the list?

**Comment by saturnonearth** [comment]
I am using the virtual list to essentially not overload the DOM with thousands of items. So it would be a scrollable list of items. The items themselves are already loaded, no need for network calls or loading.

The issues that arise in my head are that svelte-dnd-action seems to require a #each list of items, but the virtual list is just a  component that does its own thing. The other issue I wonder about is how svelte-dnd-action deals with items that appear and disappear within the DOM.

What are your thoughts on these?

**Comment by isaacHagoel** [comment]
true. the lib abstract the list away. I would start by cloning it and trying to add svelte-dnd-action manually into its code and see if it works out of the box. if yes, the next step would be to talk to the maintainer to see if they are open to a PR that adds a new option that enables dnd for the virtual list.

**Comment by saturnonearth** [comment]
I did as you suggested - it appears that it does not work 100% because it thinks the list ends earlier than it really does.

**Comment by isaacHagoel** [comment]
can you pls add a video or link to somewhere (e.g code with instructions
for how to run locally)?

**Comment by saturnonearth** [comment]
I created a repo using 2 different examples, the 1st being using the native svelte-tiny-virtual-list without modifications. The second example I extracted the component out and added svelte-dnd directly to it so you have access to the #each block. The second example is the one that "sort of" works. But as you can see via this GIF:

![chrome_04uWqfsCLh](https://user-images.githubusercontent.com/12503538/218818713-1e23cf21-5194-4e4d-a803-e773c034f09d.gif)

You cannot seem to be able to drag items beneath the set initial height (indicated by the yellow line) from the virtual list.

Here is the repo:
https://github.com/saturnonearth/virtual-dnd

**Comment by isaacHagoel** [comment]
i didn't look at the repo yet but based on the video the items are overflowing from the container. see that yellow line?
can you make the container wrap around the items?

**Comment by asahelk** [comment]
I created a playground example with [https://www.npmjs.com/package/svelte-tiny-virtual-list](url)

[https://svelte.dev/playground/84c59f1d6cae46e78002fb263e128ecf](url)

**Comment by isaacHagoel** [comment]
Closing as an integration/support question. Virtual lists render only part of the items array and therefore do not naturally satisfy the dndzone child/item contract; the discussion contains community examples for consumers who want to combine them.
